.github/workflows/deploy.yml

```diff
@@ -0,0 +1,48 @@
+name: Deploy
+
+on:
+  push:
+    branches:
+      - dev
+
+jobs:
+  cdk-deploy:
+    runs-on: ubuntu-latest
+    # These permissions are needed to interact with GitHub's OIDC Token endpoint.
+    permissions:
+      id-token: write
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - uses: Gr1N/setup-poetry@v8
+
+      - name: Install dependencies and app
+        run: |
+          poetry install --with lambda-dev
+
+      - name: Install cdk
+        run: |
+          npm install -g aws-cdk@2.100.0
+
+      # https://github.com/aws-actions/configure-aws-credentials
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v4
+        with:
+          role-to-assume: arn:aws:iam::449431850278:role/GitHubDeploy
+          aws-region: us-west-2
+
+      - name: Synth
+        run: |
+          # Activate the poetry virtualenv where we installed our package
+          source $(poetry env info --path)/bin/activate
+          cdk synth --context account_name=dev
```

Comment:
How do you imagine eventually handling the account name in these deploys? Not super relevant for this PR, just curious if you have a vision

Reply:
I was thinking we would make different steps for these cases and put an `if: branch_name == "main"` or `if: branch_name == "dev"`. But, I haven't thought real hard about this. I guess we could just put in `account_name=${GITHUB_REF##*/}` to get the git branch name to use. Then we'd need a dev and prod branch on this project for deploys.